src/dvc_data/hashfile/checkout.py

```diff
@@ -158,9 +162,6 @@ def __init__(self, links, callback: "Callback" = DEFAULT_CALLBACK):
         self._callback = callback
 
     def __call__(self, cache, from_path, to_fs, to_path):
-        if to_fs.exists(to_path):
-            to_fs.remove(to_path)  # broken symlink
-
```

Comment:
`Link.__call__` is called right after `_remove`. 

https://github.com/iterative/dvc-data/blob/3b53bd2a8103d69550cd0f7c435ddb4e76bf8d84/src/dvc_data/hashfile/checkout.py#L59-L60